src/keytoscript.js

```diff
@@ -0,0 +1,124 @@
+var bcrypto = require('./crypto')
+var btemplates = require('./templates')
+
+function checkAllowedP2sh (keyFactory) {
```

Comment:
these two functions suck, doesn't really allow people to bring their own functions.. but wanted run-time prevention against nesting script hash types incorrectly.